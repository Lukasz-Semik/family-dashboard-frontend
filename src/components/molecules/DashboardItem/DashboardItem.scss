.item-wrapper {
  width: 45rem;

  img {
    display: block;
    width: 7rem;
    height: 7rem;
    margin-right: 1rem;
    border-radius: 50%;
  }
}

.header-group {
  @include flex-center();

  padding: 5rem 0;
}

.item-inner-wrapper {
  padding: 0 3rem;
  min-height: 43rem;
}

.items-description {
  margin-bottom: 3rem;
}

.items-counter {
  margin: 1rem 0;
}

.link-wrapper {
  @include flex-center-h();

  margin-top: 3rem;
  margin-bottom: 4rem;
}
