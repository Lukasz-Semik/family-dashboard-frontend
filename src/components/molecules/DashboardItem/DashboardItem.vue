<template>
  <div :class="[$style['item-wrapper']]">
    <CardElement>
      <div :class="[$style['item-inner-wrapper']]">
        <div :class="[$style['header-group']]">
          <img :src="photoSrc">

          <TitleElement tag="h2" :translationPath="itemTitle" is-orange has-centered-text/>
        </div>

        <div :class="[$style['items-description']]">
          <TextElement :translationPath="itemDescription" is-medium has-centered-text/>
        </div>

        <div :class="[$style['items-counter']]">
          <TextElement
            translationPath="dashboard.allItems"
            :translationValues="{quantity: 0}"
            is-bold
            has-centered-text
          />
        </div>

        <div :class="[$style['items-counter']]">
          <TextElement
            translationPath="dashboard.notDoneItems"
            :translationValues="{quantity: 0}"
            is-bold
            has-centered-text
          />
        </div>

        <div :class="[$style['link-wrapper']]">
          <Link :to="destinationRoute" translation-path="dashboard.nav.goThere" is-x-big/>
        </div>
      </div>
    </CardElement>
  </div>
</template>

<script>
import CardElement from '@/components/atoms/CardElement/CardElement.vue';
import TitleElement from '@/components/atoms/TitleElement/TitleElement.vue';
import TextElement from '@/components/atoms/TextElement/TextElement.vue';
import Link from '@/components/atoms/Link/Link.vue';
// TODO: alt for images!
export default {
  components: {
    CardElement,
    TitleElement,
    TextElement,
    Link,
  },
  props: {
    itemTitle: {
      type: String,
      default: '',
    },
    itemDescription: {
      type: String,
      default: '',
    },
    photoSrc: {
      type: String,
      defautl: '',
    },
    destinationRoute: {
      type: Object,
      default: () => {},
    },
  },
};
</script>

<style lang="scss" module src="./DashboardItem.scss" />
