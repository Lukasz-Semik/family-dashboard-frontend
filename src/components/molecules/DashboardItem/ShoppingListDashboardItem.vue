<template>
  <DashboardItem
    item-title="shoppingLists.title"
    item-description="shoppingLists.description"
    :destination-route="shoppingListsRoute"
    :photo-src="shoppingListPhoto"
  />
</template>

<script>
import shoppingListPhoto from '@/assets/shoppinglist.png';
import { SHOPPING_LISTS_ROUTE } from '@/constants/routesNames';

import DashboardItem from './DashboardItem.vue';

export default {
  components: {
    DashboardItem,
  },
  data() {
    return {
      shoppingListsRoute: { name: SHOPPING_LISTS_ROUTE },
      shoppingListPhoto,
    };
  },
};
</script>
