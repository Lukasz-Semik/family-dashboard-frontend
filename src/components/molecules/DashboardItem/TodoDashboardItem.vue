<template>
  <DashboardItem
    item-title="todos.title"
    item-description="todos.description"
    :destination-route="todosRoute"
    :photo-src="todoListPhoto"
  />
</template>

<script>
import todoListPhoto from '@/assets/todolist.jpg';
import { TODOS_ROUTE } from '@/constants/routesNames';

import DashboardItem from './DashboardItem.vue';

export default {
  components: {
    DashboardItem,
  },
  data() {
    return {
      todosRoute: { name: TODOS_ROUTE },
      todoListPhoto,
    };
  },
};
</script>
