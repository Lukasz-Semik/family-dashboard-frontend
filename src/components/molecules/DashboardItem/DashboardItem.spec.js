import { shallowMount } from '@vue/test-utils';

import DashboardItem from './DashboardItem.vue';

describe('<DashboardItem />', () => {
  const props = {
    itemTitle: 'itemTitle.path',
    itemDescription: 'itemDescription.path',
    photoSrc: 'some.img',
    destinationRoute: { name: '/some-path' },
  };

  const wrapper = shallowMount(DashboardItem, {
    propsData: { ...props },
    mocks: {
      $t: key => key,
    },
  });

  it('should', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
