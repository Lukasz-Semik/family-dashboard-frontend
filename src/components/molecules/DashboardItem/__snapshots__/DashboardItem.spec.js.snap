// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DashboardItem /> should 1`] = `
<div class="item-wrapper">
  <cardelement-stub ishoverable="true">
    <div class="item-inner-wrapper">
      <div class="header-group"><img src="some.img" alt="itemTitle.path">
        <titleelement-stub tag="h2" translatedtext="" translationpath="itemTitle.path" isorange="true" hascenteredtext="true"></titleelement-stub>
      </div>
      <div class="items-description">
        <textelement-stub translatedtext="" translationpath="itemDescription.path" hascenteredtext="true" ismedium="true"></textelement-stub>
      </div>
      <div class="items-counter">
        <textelement-stub translatedtext="" translationpath="dashboard.allItems" translationvalues="[object Object]" hascenteredtext="true" isbold="true"></textelement-stub>
      </div>
      <div class="items-counter">
        <textelement-stub translatedtext="" translationpath="dashboard.notDoneItems" translationvalues="[object Object]" hascenteredtext="true" isbold="true"></textelement-stub>
      </div>
      <div class="link-wrapper">
        <link-stub to="[object Object]" translatedtext="" translationpath="dashboard.nav.goThere" isxbig="true"></link-stub>
      </div>
    </div>
  </cardelement-stub>
</div>
`;
